pkg/kudoctl/cmd/package_new.go

```diff
@@ -0,0 +1,147 @@
+package cmd
+
+import (
+	"errors"
+	"io"
+
+	"github.com/Masterminds/semver"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+
+	"github.com/kudobuilder/kudo/pkg/kudoctl/cmd/generate"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/cmd/prompt"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/packages"
+	"github.com/kudobuilder/kudo/pkg/version"
+)
+
+const (
+	pkgNewDesc = `Create a new KUDO operator on the local filesystem`
+
+	pkgNewExample = `  # Create a new KUDO operator name foo 
+  kubectl kudo package foo
+`
+)
+
+type packageNewCmd struct {
+	name        string
+	out         io.Writer
+	fs          afero.Fs
+	interactive bool
+	overwrite   bool
+}
+
+// newPackageNewCmd creates an operator package on the file system
+func newPackageNewCmd(fs afero.Fs, out io.Writer) *cobra.Command {
+
+	pkg := &packageNewCmd{out: out, fs: fs}
+	cmd := &cobra.Command{
+		Use:     "new <operator name>",
+		Short:   "create new operator",
+		Long:    pkgNewDesc,
+		Example: pkgNewExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := validateOperatorArg(args); err != nil {
+				return err
+			}
+			pkg.name = args[0]
+			if err := pkg.run(); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+
+	f := cmd.Flags()
+	f.BoolVarP(&pkg.interactive, "interactive", "i", false, "Interactively create operator")
+	f.BoolVarP(&pkg.overwrite, "overwrite", "w", false, "overwrite existing directory and operator.yaml file")
+	return cmd
+}
+
+// run returns the errors associated with cmd env
+func (pkg *packageNewCmd) run() error {
+	folderDefault := "operator"
+	ovDefault := "0.1.0"
+	kudoDefault := version.Get().GitVersion
+	apiVerDefault := "kudo.dev/v1beta1"
+
+	if !pkg.interactive {
+		op := packages.OperatorFile{
+			Name:        pkg.name,
+			APIVersion:  apiVerDefault,
+			Version:     ovDefault,
+			KUDOVersion: kudoDefault,
+		}
+
+		return generate.Operator(pkg.fs, folderDefault, op, pkg.overwrite)
+	}
+
+	// interactive mode
+	nvalid := func(input string) error {
+		if len(input) < 3 {
+			return errors.New("Operator name must have more than 3 characters")
+		}
+		return nil
+	}
+
+	name, err := prompt.WithValidator("Operator Name", pkg.name, nvalid)
+	if err != nil {
+		return err
+	}
+
+	dvalid := func(input string) error {
+		if len(input) < 1 {
+			return errors.New("Operator directory must have more than 1 character")
+		}
+		return generate.OperatorCheck(pkg.fs, input, pkg.overwrite)
+	}
+
+	dir, err := prompt.WithValidator("Operator directory", folderDefault, dvalid)
+	if err != nil {
+		return err
+	}
+
+	//TODO (kensipe): need list of supported versions some where
+	vOptions := []string{"kudo.dev/v1beta1"}
+	apiVersion, err := prompt.WithOptions("API Version", vOptions)
```

Comment:
let's not make api versions optional at all. I think even when we have multiple, we always want the new operators to come in the newest version so I would not even give them the choice to choose...

Reply:
and especially right now when we have one, it's just an extra step that can be confusing to someone

Reply:
yeah... that makes a lot of sense!  thanks!